moneypy/importers/importer.py

```
@@ -0,0 +1,14 @@
+import pandas as pd
+import regex as re
+
+def find_fields(*args, **kwargs):
+    file = pd.read_csv(*args)
+    fields = []
+
+    if not kwargs: kwargs = dict(date = 'date', amount = 'amount', desc = 'description')
+
+    for v in kwargs.items():
+        f = file[[col for col in file.columns.values if re.search(v[1], col, re.IGNORECASE)]]
```

While searching the columns with regex is super cool, I think it might be too error prone. I would say, at this point we should just force the user to specify the full and unique column name.

With that the filtering could work something like this:

- get the values from the column_dict
- get all columns belonging to these values
- Invert the dictionary and use it to rename the columns you have extracted

These 3 things together should be 2 very simple lines :) If you want to try on your own, go for it, but if you need "inspiration" let me know.

---

At this point we can also raise an error if one of our required columns don't exist